src/ui/pages/Login/Login.tsx

```diff
@@ -12,10 +19,78 @@ import { VStack } from 'src/ui/ui-kit/VStack';
 import * as s from 'src/ui/style/helpers.module.css';
 import { UnstyledLink } from 'src/ui/ui-kit/UnstyledLink';
 import { Input } from 'src/ui/ui-kit/Input';
-import { HStack } from 'src/ui/ui-kit/HStack';
 import ZerionLogo from 'jsx:src/ui/assets/zerion-squircle.svg';
 import { useBodyStyle } from 'src/ui/components/Background/Background';
 import { zeroizeAfterSubmission } from 'src/ui/shared/zeroize-submission';
+import { PageFullBleedColumn } from 'src/ui/components/PageFullBleedColumn';
+import { WalletAvatar } from 'src/ui/components/WalletAvatar';
+import { ZStack } from 'src/ui/ui-kit/ZStack';
+import { walletPort } from 'src/ui/shared/channels';
+import { invariant } from 'src/shared/invariant';
+import { DelayedRender } from 'src/ui/components/DelayedRender';
+import { LayersAnimationLottie } from './LayersAnimationLottie';
+import { type LottieComponentHandle } from './LayersAnimationLottie';
+
+function LoginPageAnimation({ address }: { address: string | null }) {
+  const [lottieIsReady, setLottieIsReady] = useState(false);
+  const [avatarImageIsReady, setAvatarImageIsReady] = useState(false);
+  const [lottieStarted, setLottieStarted] = useState(false);
+  const handleAvatarImageReady = useCallback(
+    () => setAvatarImageIsReady(true),
+    []
+  );
+  const lottieComponentRef = useRef<LottieComponentHandle | null>(null);
+  const hasAddressValue = Boolean(address);
+  useEffect(() => {
+    if (lottieIsReady && (!hasAddressValue || avatarImageIsReady)) {
+      lottieComponentRef.current?.startAnimation();
+      setLottieStarted(true);
+    }
+  }, [lottieIsReady, avatarImageIsReady, hasAddressValue]);
+  return (
+    <div style={{ position: 'relative' }}>
+      <div style={{ position: 'absolute' }}>
+        <DelayedRender>
+          {/* Expensive component, delay rendering so that initial view render is fast */}
+          <LayersAnimationLottie
+            ref={lottieComponentRef}
+            onAnimationReady={() => setLottieIsReady(true)}
+          />
+        </DelayedRender>
+      </div>
+      <Spacer height={130} />
+      <div style={{ display: 'flex', justifyContent: 'center' }}>
+        <ZStack style={{ placeItems: 'center' }}>
+          <ZerionLogo style={{ width: 54, height: 54 }} />
+          {address ? (
+            <div
```

Comment:
Why not use react-spring here?

Reply:
Trying to experiment with new CSS features